### ISSUE TITLE
Fix *-kyma-gardener-gcp-eventing-upgrade test

### ISSUE THREAD

**Issue by nachtmaar** [kyma-project/kyma#13395]
## Description

**Reasoning**
We want to ensure that everything which can be tested in an `automated way` is `tested` and we have `less` `manual testing` in place.
As soon as https://github.com/kyma-project/kyma/issues/13393 is implemented all our manual tests cases except upgrade tests are covered. In order to also get rid of `manual` `upgrade testing` we should make use of the `kyma-gardener-gcp-eventing-upgrade` test.

Before we can convert the optional job [`kyma-gardener-gcp-eventing-upgrade`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma/kyma-integration-gardener.yaml#L386) into a mandatory job, we have to ensure that it is green most of the time and only fails if a proper error occurs

## Acceptance
- [x] investigate into the failing job history and fix if needed
- [x] test must be stable when no change to any kyma component was introduced
- [x] convert into mandatory job as soon as its execution stabilised 
- [ ] add as pre submit also to the release branches (starting latest release)

## See Also

- [pre-main-kyma-gardener-gcp-eventing-upgrade on prow status](https://status.build.kyma-project.io/?job=pre-main-kyma-gardener-gcp-eventing-upgrade)
- [kyma-gardener-gcp-eventing-upgrade prow job definition](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma/kyma-integration-gardener.yaml#L386)]

**Comment by VladislavPaskar** [comment]
Please consider this as a follow-up issue: https://github.com/kyma-project/kyma/issues/13520

**Comment by FriedrichWilken** [comment]
The job runs the tests from main against an `2.0.4` cluster where they fail, before the actual upgrade.
This is reproducible by manually executing the tests against a `kyma 2.0.04` with `make ci-test-eventing` from /kyma/tests/fast-integration (`main`).

The tests fail with 
```
 1) Eventing tests
       with Nats backend
         In-cluster event should have correct tracing spans:
     Error: cannot get jaeger trace
404: Not Found: {"data":null,"total":0,"limit":0,"offset":0,"errors":[{"code":404,"msg":"trace not found"}]}
      at convertAxiosError (utils/index.js:109:10)
      at getJaegerTrace (monitoring/client.js:132:11)
      at processTicksAndRejections (internal/process/task_queues.js:95:5)
      at async checkTrace (test/fixtures/commerce-mock/index.js:334:16)
      at async checkInClusterEventTracing (test/fixtures/commerce-mock/index.js:326:3)
      at async Context.<anonymous> (eventing-test/eventing-test.js:99:7)
```

[This function](https://github.com/kyma-project/kyma/blob/b4b0d56d1d3c0c9d2b10b15f63be177fd8db332b/tests/fast-integration/test/fixtures/commerce-mock/index.js#L829) will return a response that contains the traceID. For some reason this ID does not exist in Jaeger. To check this, set a breakpoint [here](https://github.com/kyma-project/kyma/blob/b4b0d56d1d3c0c9d2b10b15f63be177fd8db332b/tests/fast-integration/test/fixtures/commerce-mock/index.js#L312) and debug the test. You can copy the traceID, and manually open Jaeger, to confirm, that it does not exist.

**Comment by marcobebway** [comment]
Blocked by: https://github.com/kyma-project/kyma/issues/13789.

**Comment by marcobebway** [comment]
The job looks green, we decided to make it required.

**Comment by nachtmaar** [comment]
@marcobebway are all the acceptance criteria implemented ? Can you please check the boxes ?

**Comment by VladislavPaskar** [comment]
I think all the acceptance criteria for this pipeline is fulfilled now. The job is pretty stable and required now. I will only move the 
```
[ ] add as pre submit also to the release branches (starting latest release)
```
to another ticket, as it doesn't have anything to do with fixes. So we can can close this issue.

**Comment by raypinto** [comment]
Making this pipeline optional again using [this PR](https://github.com/kyma-project/test-infra/pull/5623), until we resolve [this issue](https://github.com/kyma-project/kyma/issues/13789). It will then be made required as a pre-requisite for [this issue](https://github.com/kyma-project/kyma/issues/14569).
